src/charm.py

```diff
@@ -386,6 +390,16 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
         self._container.restart(self._container_name)
         self.unit.status = ActiveStatus()
 
+    def _on_install(self, event: InstallEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Hydra container. Deferring event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
+            return
+
+        if not self._container.isdir(self._log_dir):
+            self._container.make_dir(path=self._log_dir, make_parents=True, permissions=0o777)
```

Comment:
I'd recommend either using persistent storage or leveraging pebble ready event. Here the make dir will only run on install but if the pod dies and gets rescheduled, this dir won't exist.

Reply:
FYI I put make_dir into the pebble ready handler. I also put a can connect to container check into the handler to make sure the container is available, and to not call the defer() method on the event twice. (once during checking for make_dir, and once during running _handle_status_update_config.